knowledge_runtime/knowledge_runtime/services/duckdb_generator.py

```
@@ -0,0 +1,656 @@
+        else:
+            # Multi-sheet: import each non-empty sheet
+            used_names: list[str] = []
+            for sheet_name in sheet_names:
+                table_name = self._sanitize_table_name(f"sheet_{sheet_name}")
+                table_name = self._deduplicate_table_name(table_name, used_names)
+                used_names.append(table_name)
+
+                try:
+                    conn.execute(
+                        f'CREATE TABLE "{table_name}" AS '
+                        "SELECT * FROM read_xlsx(?, header=true, sheet=?)",
```

Ruff (0.15.12)

[error] Possible SQL injection vector through string-based query construction

(S608)